x-pack/metricbeat/module/azure/billing/data.go

```diff
@@ -60,6 +60,10 @@ func EventsMapping(
 						"group": legacy.Properties.ResourceGroup,
 					},
 				}
+				if len(legacy.Tags) > 0 {
+					_, _ = event.ModuleFields.Put("resource.tags", legacy.Tags)
```

Comment:
Should we be capturing and checking the returned error value here, like we do here? I see that it's being ignored at most places, I am okay if we do it here as well seeing the urgency of the issue.

Reply:
@devamanv In that case `node.name` seems like a more important piece of information that must be present, not sure if it makes sense to continue here if adding `resource.tags` fails, maybe logging? I might be wrong though.

Reply:
@gpop63 The idea is to just make sure we are handling the error in some way. I mentioned that line just for a reference, and not to do the same thing here.

Reply:
My understanding of checking or not the calls to the `Put() 's on `mapstr.M` type is how confident we are the map can store actual values passed. 

The `Put()` function has this signature:

```golang
Put(key string, value interface{}) (interface{}, error)
```

In this case, the value we pass to the `value` argument has the `map[string]*string` type.  By looking at the `Put()` [implementation](https://github.com/elastic/elastic-agent-libs/blob/4dea101389eae1e588e9d87aab5abcbb51171cee/mapstr/mapstr.go#L195), I don't see potential issues for the `map[string]*string` type.

The last time I checked, I saw both error-checked and muted use of `Put()` calls, depending on the context.

Reply:
For example, I put together a small test at https://go.dev/play/p/PvA_LzSi7sh with this source code:

```golang
// You can edit this code!
// Click here and start typing.
package main

import (
	"fmt"

	"github.com/elastic/elastic-agent-libs/mapstr"
)

func main() {

	m := mapstr.M{
		"resource": mapstr.M{},
	}

	//
	// Empty
	//

	_, _ = m.Put("resource.tags", map[string]*string{})
	fmt.Println("Hello, empty tags!", m)

	//
	// With values
	//

	greeting := "dude"

	tags := map[string]*string{
		"hey-with-nil":    nil,
		"hey-with-string": &greeting,
	}

	_, _ = m.Put("resource.tags", tags)
	fmt.Println("Hello, tags!", m)
}
```

That produces this output:

```text
Hello, empty tags! {"resource":{"tags":{}}}
Hello, tags! {"resource":{"tags":{"hey-with-nil":null,"hey-with-string":"dude"}}}
```

The `map[string]*string` seems pretty safe. 

The only way I can mess up is by calling `Put()` with a different `key` value, but this is something unit tests can handle.

Reply:
Fair enough. I am good, resolving the comment.